### PR TITLE
feat(unstable): kv.watch()

### PR DESCRIPTION
**Pull request by lucacasonato**
This commit adds support for a new `kv.watch()` method that allows
watching for changes to a key-value pair. This is useful for cases
where you want to be notified when a key-value pair changes, but
don't want to have to poll for changes.

### CODE REVIEW

File: ext/kv/lib.rs [hunk 1]
```diff
@@ -354,6 +381,86 @@ where
...
+) -> Result<ResourceId, AnyError>
+where
+  DBH: DatabaseHandler + 'static,
+{
+  let resource = state.resource_table.get::<DatabaseResource<DBH::DB>>(rid)?;
+
+  let keys = keys
+    .into_iter()
+    .map(encode_v8_key)
+    .collect::<std::io::Result<_>>()?;
```
Comment (igorzi) [inline ext/kv/lib.rs:399]: There should be a limit on the number of keys to watch

Comment (losfair) [reply ext/kv/lib.rs]: Added

**Comment by bartlomieju** [comment]
Please land it as `feat(unstable): kv.watch()`

**Review by losfair — APPROVED** [review]

**Comment by thegaryroberts** [comment]
image

**Pull request merged**